mlir/lib/Dialect/Rock/Transforms/AlignTiling.cpp

```diff
@@ -373,11 +385,57 @@ static bool isUnfusedKernelStore(ThreadwiseWriteAllOp store) {
   return ret;
 }
 
+static SmallVector<Attribute> extractVector(ArrayAttr arrayAttr) {
+  return llvm::to_vector<4>(arrayAttr.getAsRange<Attribute>());
+}
+
+LogicalResult
+ReduceRewritePattern::matchAndRewrite(rock::ReduceOp reduceOp,
+                                      PatternRewriter &rewriter) const {
+  Location loc = reduceOp.getLoc();
+  ThreadwiseWriteAllOp threadwiseWriteOp =
+      traceToThreadwiseWrite(reduceOp.getIn());
+  if (reduceOp.getReduceMethod() != ReduceMethod::Sum) {
```

Comment:
I think this is a `matchFailure()`

Reply:
I think failing to match the pattern(rewrite) doesn't mean some other rewriter is going to take care of that in this case. it will just make the error more obscure...

because at this stage this failure means compilation failure. I think we should fail here if it comes here because we already committed to fusion by now.